### DISCUSSION TITLE
Using webhook on local network

### DISCUSSION THREAD

**Discussion by k9protection** [capcom6/android-sms-gateway#87] [Q&A] [answered]
Hi, firstly, please excuse my naivety here - my programming skills stopped at vb6. I've currently got a lad working on writing an ASP page to register a webhook and receive messages from the phone running the SMS gateway. He has informed me that the machine registering the webhook must be SSL and must be on the internet.

A couple of questions if I may:

Our servers have an SSL wildcard certificate, will this suffice or would they need a dedicated certificate for the server? (now is a good time as it's certificate renewal month).

Secondly, does the receiving server need to be on the internet, or will having an FQDN that resolves to the server be sufficient? Our usage case is for internal use and sending messages to staff and allowing them to reply.. I don't want to have to have a page that can POST to our internal staff database as web facing if I can avoid it.

Cheers all

Oh, and thank you for developing this :)

**Comment by capcom6** [comment, marked as answer]
Hello, @k9protection !

1. You need a valid certificate that your Android device can trust. To test this, simply try to open your endpoint in the default browser on your device. If there is no warning about the certificate, it will work. You can even use a self-signed certificate if you add the CA cert to the trust list on your device - look for the Security section in your device settings.

2. Because webhooks are sent directly from the device, it is not necessary for your server to have access to the Internet. In this case, your server must be accessible from the device, for example, by connecting to the same network. The easiest way to test this is similar to point 1 - try to open your server's address from a browser on the device.

Thank you for your interest in this project!
